site/app/libraries/database/DatabaseQueries.php

```diff
@@ -183,6 +183,23 @@ public function buildLoadThreadQuery($categories_ids, $thread_status, $show_dele
 
             $query_raw_join[] = "JOIN ({$query_select_categories}) AS QSC ON QSC.thread_id = t.id";
         }
+        // Unread Threads
+        if($unread_threads) {
+            $query_raw_where[] = 
+
+            "EXISTS(
+                SELECT thread_id 
+                FROM (posts LEFT JOIN forum_posts_history ON posts.id = forum_posts_history.post_id) AS jp
+                WHERE(
+                    jp.thread_id = t.id
+                    AND NOT EXISTS(
+                        SELECT thread_id 
+                        FROM viewed_responses v 
+                        WHERE v.thread_id = jp.thread_id
+                            AND v.user_id = '{$current_user}'
```

Comment:
Added $current_user to the array $query_parameters. I've noticed are there are other times in buildLoadThreadQuery where php strings are put straight into queries. When should we use the '?' instead of the direct string placement?

Reply:
Generally never for stuff that're your testing against/setting in the query, especially anything that could be set by an end-user. The only real reason to have PHP variables within a query is because you're dynamically putting together a query (like there's a dynamic number of `?` within an insert), but otherwise use the `?` syntax to avoid potential SQL injections, even if you're totally positive that variable is sanitized and cannot be manipulated as a string by an end user.